crates/optimism/primitives/src/receipt.rs

```diff
@@ -211,6 +211,21 @@ impl InMemorySize for OpReceipt {
 
 impl reth_primitives_traits::Receipt for OpReceipt {}
 
+/// Trait for deposit receipt.
+pub trait DepositReceipt: reth_primitives_traits::Receipt {
```

Comment:
I think we can live with this returning Option and we expect that this will be implemented for the actual receipt enum, so no need to to also name this `OpReceipt`